src/transformers/models/parakeet/modular_parakeet.py

```diff
@@ -483,6 +483,277 @@ class ParakeetGenerateOutput(ModelOutput):
+        )
+
+        if t_max is not None:
+            # apply chrono init
+            for name, v in self.lstm.named_parameters():
+                if 'bias' in name:
+                    p = getattr(self.lstm, name)
+                    n = p.nelement()
+                    hidden_size = n // 4
+                    p.data.fill_(0)
+                    p.data[hidden_size : 2 * hidden_size] = torch.log(
+                        torch.nn.init.uniform_(p.data[0:hidden_size], 1, t_max - 1)
+                    )
+                    # forget gate biases = log(uniform(1, Tmax-1))
+                    p.data[0:hidden_size] = -p.data[hidden_size : 2 * hidden_size]
+                    # input gate biases = -(forget gate biases)
+
+        elif forget_gate_bias is not None:
+            for name, v in self.lstm.named_parameters():
+                if "bias_ih" in name:
+                    bias = getattr(self.lstm, name)
+                    bias.data[hidden_size : 2 * hidden_size].fill_(forget_gate_bias)
+                if "bias_hh" in name:
+                    bias = getattr(self.lstm, name)
+                    bias.data[hidden_size : 2 * hidden_size] *= float(hidden_hidden_bias_scale)
+
+        self.dropout = torch.nn.Dropout(dropout) if dropout else None
+
+        for name, v in self.named_parameters():
+            if 'weight' in name or 'bias' in name:
+                v.data *= float(weights_init_scale)
+
+    def forward(
+        self, x: torch.Tensor, h: Optional[Tuple[torch.Tensor, torch.Tensor]] = None
+    ) -> Tuple[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+
+        x, h = self.lstm(x, h)
+
+        if self.dropout:
+            x = self.dropout(x)
+
+        return x, h
+
+class ParakeetTDTJoint(ParakeetPreTrainedModel):
+    config: ParakeetTDTJointConfig
+    base_model_prefix = "" #joint"
+    main_input_name = "enc"
+    _supports_flat_attention_mask = False
+    _supports_sdpa = True
+    _supports_flex_attn = False
+    _supports_attention_backend = False
+    _can_record_outputs = {}
+    _no_split_modules = None
+
+    def __init__(self, config: ParakeetTDTJointConfig):
+        super().__init__(config)
+        self.config = config
+        self.gradient_checkpointing = False
+
+        self.enc = torch.nn.Linear(config.enc_hidden_size, config.hidden_size)
+        self.pred = torch.nn.Linear(config.pred_hidden_size, config.hidden_size)
+
+        num_classes = config.vocab_size + 1 + len(config.durations)
+
+        layers = (
+            [torch.nn.ReLU(inplace=True)]
+            + ([torch.nn.Dropout(p=self.config.dropout)])
+            + [torch.nn.Linear(config.hidden_size, num_classes)]
+        )
+        self.joint_net = torch.nn.Sequential(*layers)
+        self.post_init()
+
+    @auto_docstring
+    @check_model_inputs()
+    def forward(
+        self,
+        enc: torch.Tensor,
+        pred: torch.Tensor,
```

Comment:
can we use more verbose variable names? e.g. `encoder_output` and `decoder_output`